containers/ecr-viewer/src/app/data/db/schemas/core/19700101000001_initial.ts

```diff
@@ -0,0 +1,62 @@
+import { Kysely, sql } from "kysely";
+
+import { dbSchema } from "@/app/api/services/database";
+import { getSql } from "@/app/api/services/dialects/common";
+
+/**
+ * Based on ecr-viewer/sql/core.sql.
+ * Core schema initialization. This version is hard-coded to the
+ * original postgres implementation. Future versions will be
+ * database-agnostic.
+ * @param db - the database connection
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function up(db: Kysely<any>): Promise<void> {
+  if (dbSchema() === "extended") {
+    console.log("Extended schema detected. Skipping core migration.");
+    return;
+  }
+
+  // Install uuid-ossp extension (Postgres-specific)
+  await sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp";`.execute(db);
+
+  const coreCheck = await db
+    .selectFrom("ecr_viewer.ecr_data")
+    .select("patient_name_first")
+    .executeTakeFirst();
+
+  if (!coreCheck) {
+    console.log("Core migration already run. Skipping table creation.");
+    return;
+  }
+
+  await db.schema
+    .alterTable("ecr_viewer.ecr_data")
+    .addColumn("data_source", "varchar(2)", (cb) => cb.notNull()) // S3 or DB
+    .addColumn("patient_name_first", "varchar(100)")
+    .addColumn("patient_name_last", "varchar(100)")
+    .addColumn("patient_birth_date", getSql("datetimeType"))
+    .addColumn("report_date", getSql("datetimeType"), (cb) => cb.notNull())
+    .execute();
+}
+
+/**
+ * Based on ecr-viewer/sql/core.sql.
+ * Core schema tear down. This version is hard-coded to the
+ * original postgres implementation. Future versions will be
+ * database-agnostic.
+ * @param db - the database connection
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function down(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .dropTable("ecr_viewer.ecr_rr_rule_summaries")
+    .ifExists()
+    .execute();
+  await db.schema
+    .dropTable("ecr_viewer.ecr_rr_conditions")
+    .ifExists()
+    .execute();
+  await db.schema.dropTable("ecr_viewer.ecr_data").ifExists().execute();
+  await db.schema.dropSchema("ecr_viewer").ifExists().execute();
```

Comment:
strictly speaking, shouldn't this remove the columns added here? or at least only drop the ecr_data?